### PR TITLE
[Fix](mluOpSetFFTReserveArea): fix fft bug

### PR DESCRIPTION
**Pull request by aokbok**
Thanks for your contribution and we appreciate it a lot. :rocket::rocket:

## 1. Motivation

Please describe your motivation and the goal you want to achieve through this pull request.

## 2. Modification

Please briefly describe what modification is made in this pull request, and indicate where to make the modification.

Are new test cases added? If so, please post the corresponding generator-PR link here.

## 3. Test Report

If you want to know how to do operator testing, you can see [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md).

#### 3.1.1 Accuracy Acceptance Standard

For static threshold standard details, see: [MLU-OPS™ Accuracy Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Accuracy-Acceptance-Standard.md).

- static threshold
  - diff1
    - [ ] float32 mlu diff1 <= 1e-5
    - [ ] float32 mlu diff1 <= 3e-3
    - [ ] float16 mlu diff1 <= 3e-3
  - diff2
    - [ ] float32 mlu diff2 <= 1e-5
    - [ ] float32 mlu diff2 <= 3e-3
    - [ ] float16 mlu diff2 <= 3e-3
  - diff3
    - [ ] mlu diff3 == 0
    - [ ] mlu diff3_1 == 0
    - [ ] mlu diff3_2 == 0
- dynamic threshold
  - [ ] diff1: mlu diff1 <= max(baseline diff1 * 10, static threshold)
  - [ ] diff2: mlu diff2 <= max(baseline diff2 * 10, static threshold)
  - [ ] diff3: mlu diff3 <= max(baseline diff3 * 10, static threshold)
    - float32, threshold = 1e-5
    - float16, threshold = 1e-3

#### 3.1.2 Operator Scheme checklist

- Supported hardware
  - [ ] MLU370
  - [ ] MLU590
- Job types
  - [ ] BLOCK
  - [ ] UNION1
  - [ ] UNION2
  - [ ] UNION4
  - [ ] The operator will dynamically select the most suitable task type, for example, UNION8

#### 3.2.1 Accuracy Test

If you have checked the following items, please tick the relevant box.

- [ ] Data type test (e.g. float32/int8)
- [ ] Multi-dimensional tensor test
- [ ] Layout test
- [ ] Different size/integer remainder end segment/alignment misalignment test
- [ ] Zero dimensional tensor test/zero element test
- [ ] stability test
- [ ] Multiple platform test
- [ ] Gen_case module test, see: [Gencase-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/Gencase-User-Guide-zh.md)
- [ ] Nan/INF tests 
- [ ] Bug fix tests
- [ ] For memory leak check details, see: [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md)
- [ ] For code coverage check details, see: [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md)
- [ ] For I/O calculation efficiency check details, see: [MLU-OPS™-Performance-Acceptance-Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Performance-Acceptance-Standard.md)

#### 3.2.2 Parameter Check

Test Point-1: `When a new operator is submitted, the test points are given and the test results are stated`. Acceptance Standard: `Normal error`.
```bash
Please fill your test results(Error Message) in here, ...
```

Test Point-2: `Whether illegal parameters are passed`. Acceptance Standard: `Normal error`.
```bash
Test results...
```

### 3.3 Performance Test

See [MLU-OPS™ Performance Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Performance-Acceptance-Standard.md) for details.

Platform：MLU370

```bash
# The test results should contain Op name, Shape, Data type,  
#   MLU Hardware Time(us), MLU Interface Time(us), MLU IO Efficiency, 
#   MLU Compute Efficiency, and Mlu Workspace Size(Bytes)
# 
# for example:
#
# ----------- case0 -----------
# case0
# [Op name                ]: abs
# [Shape                  ]: input.shape=[1024,1024,3,4], output.shape=[1024,1024,3,4]
# [Data type]             ]: float32
# [MLU Hardware Time      ]: 15728 (us)
# [MLU Interface Time     ]: 369.008 (us)
# [MLU IO Efficiency      ]: 0.23275
# [MLU Compute Efficiency ]: 0.5
# [Mlu Workspace Size     ]: -1 (Bytes)
# 
# ----------- case1 -----------
# ...
```

Platform：MLU590
```bash
# ----------- case0 -----------
# ----------- case1 -----------
# ...
```

### 3.4 Summary Analysis

Please give a brief overview here, if you want to note and summarize the content.

### CODE REVIEW

**Review by PetrelYy — APPROVED** [review]

**Comment by PetrelYy** [comment]
LGTM

File: kernels/utils/common.h [hunk 1]
```diff
@@ -711,4 +712,196 @@ __mlu_vector__ void __mluop_get_indices(float *dst, float start_index,
...
+    __vv_store(dst_nram + 3 * vv_num, vv_index[3], vv_num);              \
+    __vv_store(dst_nram + 4 * vv_num, vv_index[4], vv_num);              \
+    __vv_store(dst_nram + 5 * vv_num, vv_index[5], vv_num);              \
+    __vv_store(dst_nram + 6 * vv_num, vv_index[6], vv_num);              \
+    __vv_store(dst_nram + 7 * vv_num, vv_index[7], vv_num);              \
+  } while (false)
+
+template <typename T>
+__mlu_vector__ void __cnnl_arange_vv_(T *dst_nram, T start_index, T step) {
+#if 613 <= _BANG_ARCH_
```
Comment (Unireverse) [inline kernels/utils/common.h:746]: 大于等于372

File: kernels/utils/common.h [hunk 2]
```diff
@@ -711,4 +712,196 @@ __mlu_vector__ void __mluop_get_indices(float *dst, float start_index,
...
 
+template <typename T>
+__mlu_func__ void __cnnl_arange_base_(T *dst_nram, uint32_t numel,
+                                      T start_index, T step) {
+  for (uint32_t i = 0; i < numel; i++) {
+    dst_nram[i] = start_index + i * step;
+  }
+}
+
+#define CNNL_ARANGE_VV_IMPL(VVType, vv_num, dst_nram, start_index, step) \
```
Comment (PetrelYy) [inline kernels/utils/common.h:723]: 这里 CNNL 改 MLUOP_

File: kernels/utils/common.h [hunk 3]
```diff
@@ -711,4 +712,196 @@ __mlu_vector__ void __mluop_get_indices(float *dst, float start_index,
...
+    __vv_store(dst_nram + 7 * vv_num, vv_index[7], vv_num);              \
+  } while (false)
+
+template <typename T>
+__mlu_vector__ void __mlu_op_arange_vv_(T *dst_nram, T start_index, T step) {
+#if 592 < _BANG_ARCH_
+  static_assert(
+      (std::is_same<T, float>::value || std::is_same<T, half>::value ||
+       std::is_same<T, int16_t>::value || std::is_same<T, uint16_t>::value),
+      "_cnnl_arange_vv type error!");
```
Comment (Unireverse) [inline kernels/utils/common.h:750]: 这个log改一下

**Review by Unireverse — APPROVED** [review]

**Comment by Unireverse** [comment]
LGTM

**Review by AndyQiao0828 — APPROVED** [review]

File: kernels/utils/common.h [hunk 4]
```diff
@@ -711,4 +712,220 @@ __mlu_vector__ void __mluop_get_indices(float *dst, float start_index,
...
+    __vv_add(vv_index[7], vv_index[0], 7 * vv_num * step);               \
+    __vv_store(dst_nram, vv_index[0], vv_num);                           \
+    __vv_store(dst_nram + vv_num, vv_index[1], vv_num);                  \
+    __vv_store(dst_nram + 2 * vv_num, vv_index[2], vv_num);              \
+    __vv_store(dst_nram + 3 * vv_num, vv_index[3], vv_num);              \
+    __vv_store(dst_nram + 4 * vv_num, vv_index[4], vv_num);              \
+    __vv_store(dst_nram + 5 * vv_num, vv_index[5], vv_num);              \
+    __vv_store(dst_nram + 6 * vv_num, vv_index[6], vv_num);              \
+    __vv_store(dst_nram + 7 * vv_num, vv_index[7], vv_num);              \
+  } while (false)
```
Comment (chqy99) [inline kernels/utils/common.h:742]: do while(false) 有什么作用

**Review by DanieeelLiu — APPROVED** [review]
LGTM

File: kernels/utils/common.h (hunk 4 above)
Comment (aokbok) [reply kernels/utils/common.h:742]: 有一些好处吧：参考
https://github.com/Cambricon/mlu-ops/pull/1120

**Pull request merged**